core/shared/src/main/scala/zio/ZManaged.scala

```diff
@@ -406,14 +406,9 @@ final case class ZManaged[-R, +E, +A](reserve: ZIO[R, E, Reservation[R, E, A]])
     ZManaged {
       Ref.make[Exit[Any, Any] => ZIO[R1, Nothing, Any]](_ => UIO.unit).map { finalizer =>
         Reservation(
-          acquire = ZIO.uninterruptibleMask { restore =>
-            for {
-              res   <- self.reserve
-              exitA <- restore(res.acquire).run
-              _     <- finalizer.set(exitU => res.release(exitU).ensuring(cleanup(exitA)))
-              a     <- ZIO.done(exitA)
-            } yield a
-          },
+          acquire = ZIO.bracketExit(self.reserve)(
```

Comment:
Curious, what's the motivation for the change? The differences are mainly correct usage of `ZIOFn` and error preservation (which shouldn't matter here because `finalizer.set` is a `UIO`)

Reply:
@iravid Less low-level code; also this is the only instance of `uninterruptibleMask(restore(_).catch)` aside of `bracketExit` in core and turns out it just duplicates it – making it use `bracketExit` also safeguards it against any changes to finalizer behavior :)

Reply:
Ok, makes sense!